vars/capsulePipelineWrapper.groovy

```diff
@@ -1,245 +1,485 @@
-
-            vegacapsule template node-sets \
-                --home-path '${env.CONFIG_HOME}' \
-                --path '${env.TEMPLATES_HOME}/data_node.full.tmpl.toml' \
-                --nodeset-group-name full \
-                --type data-node \
-                --with-merge ${env.FLAGS}
-            """
+/* groovylint-disable DuplicateMapLiteral, DuplicateStringLiteral, LineLength, MethodSize, NestedBlockDepth */
+void templateConfigs(String envName, String netHome, String tmplHome, String additionalFlags) {
+  sh label: 'generate templates: ' + envName, script: """
+        vegacapsule template genesis \
+            --home-path '""" + netHome + """' \
+            --path '""" + tmplHome + """/genesis.tmpl.json' """ + additionalFlags + """
+
+        vegacapsule template node-sets \
+            --home-path '""" + netHome + """' \
+            --path '""" + tmplHome + """/vega.validators.tmpl.toml' \
+            --nodeset-group-name validators \
+            --type vega \
+            --with-merge """ + additionalFlags + """
+
+        vegacapsule template node-sets \
+            --home-path '""" + netHome + """' \
+            --path '""" + tmplHome + """/tendermint.common.tmpl.toml' \
+            --nodeset-group-name validators \
+            --type tendermint \
+            --with-merge """ + additionalFlags + """
+
+        vegacapsule template node-sets \
+            --home-path '""" + netHome + """' \
+            --path '""" + tmplHome + """/vega.full.tmpl.toml' \
+            --nodeset-group-name full \
+            --type vega \
+            --with-merge """ + additionalFlags + """
+
+        vegacapsule template node-sets \
+            --home-path '""" + netHome + """' \
+            --path '""" + tmplHome + """/tendermint.common.tmpl.toml' \
+            --nodeset-group-name full \
+            --type tendermint \
+            --with-merge """ + additionalFlags + """
+
+        vegacapsule template node-sets \
+            --home-path '""" + netHome + """' \
+            --path '""" + tmplHome + """/data_node.full.tmpl.toml' \
+            --nodeset-group-name full \
+            --type data-node \
+            --with-merge """ + additionalFlags
+}
+
+void getReleasedBinary(String ghCredentialsID, String binType, String version) {
+  withGHCLI('credentialsId': ghCredentialsID) {
+        sh 'gh release --repo vegaprotocol/vega download ' + version + ' --pattern "' + binType + '-linux*"'
+  }
+
+  sh 'mv vega-linux-amd64 bin/' + binType
+  sh 'chmod +x bin/' + binType
+  sh binType + ' version'
+}
+
+void compileCoreBinary(String ghCredentialsID, String binType, String version, String outDir) {
+  Map packagesMapping = [
+        'vega': 'cmd/vega/',
+        'vegawallet': 'cmd/vegawallet/',
+        'data-node': 'cmd/data-node/'
+    ]
+
+  if (!packagesMapping.containsKey(binType)) {
+    error('[compileBinary] invalid binary type. got: "' + binType + '"')
+  }
+
+  gitClone([
```

Comment:
Nit: you don't need to add [ ] when calling function which takes only map as argument